src/Orleans/Messaging/GatewayConnection.cs

```diff
@@ -189,7 +189,7 @@ public void Connect()
         protected override bool PrepareMessageForSend(Message msg)
         {
             // Check to make sure we're not stopped
-            if (Cts.IsCancellationRequested)
+            if (Cts!=null && Cts.IsCancellationRequested)
```

Comment:
As we discussed in person, I'm adding the return on null, since no additional code should be executed in that case.